okonomiyaki/platforms/platform_filters.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import absolute_import
+
+from okonomiyaki.bundled.traitlets import (HasTraits, Bool, Enum, Instance,
+                                           List, Unicode)
+from okonomiyaki.errors import OkonomiyakiError
+
+from .platform import DARWIN, LINUX, SOLARIS, WINDOWS
+from .platform import CENTOS, RHEL, DEBIAN, UBUNTU, MAC_OS_X
+from .platform import Arch, Platform
+
+
+class PlatformLabel(HasTraits):
+    """
+    A platform filter.
+    """
+
+    os = Enum([WINDOWS, LINUX, DARWIN, SOLARIS, None])
+    """
+    The most generic OS description
+    """
+
+    name = Enum([WINDOWS, CENTOS, RHEL, DEBIAN, UBUNTU, MAC_OS_X, SOLARIS,
+                 None])
+    """
+    The most specific platform description
+    """
+
+    family = Enum([WINDOWS, RHEL, DEBIAN, MAC_OS_X, SOLARIS, None])
+    """
+    The 'kind' of platforms. For example, both debian and ubuntu distributions
+    share the same kind, 'debian'.
+    """
+
+    release = Unicode()
+    """
+    The release string. May be empty
+    """
+
+    arch = Instance(Arch)
+    """
+    Actual architecture. May be None.
+    """
+
+    @classmethod
+    def _from_legacy_string(cls, s):
+        """ Create a filter from a simple pisi string, e.g. '64', 'win-32' or
+        'all'.
+        """
+        def _epd_name_to_quadruplet(name):
+            if name == "rh":
```

Comment:
strictly speaking, one could get some other platforms starting with `rh`. As you can see in the epd_repo code, most of it is fairly ad-hoc, so I prefer to be quite conservative in what I accept.